### PR TITLE
Fix duplicated chilren in Scene spawn

### LINKED ISSUE #889: Children is been added twice
**Bevy version**
0.3 (commit fae628797fb2dba7a95a323c44c0118875b87f1c)

**Operating system & version**
Fedora 31

**What you did**
Spawn a scene with a hierarchy
```
let mut world = World::default();
let world_builder = &mut world.build();

let components = (
    GlobalTransform::default(),
    Transform::default(),
    Children::default(),
);
world_builder
    .spawn(components.clone())
    .with_children(|builder| {
        builder.spawn(components.clone()).with_children(|builder| {
            builder.spawn(components.clone()).with_children(|builder| {
                builder.spawn(components.clone()).with_children(|builder| {
                    builder.spawn(components.clone()).with_children(|builder| {
                        builder.spawn(components.clone());
                    });
                });
            });
        });
    });

commands.spawn_scene(scenes.add(Scene::new(world)));
```

**What actually happened**
Each children will be added twice to it's parent `Children` component, this causes a massive slowdown caused by the `transform_propagate_system` witch will then need to update each node hierarchy multiple times.

**Additional information**
I propose 3 things:
1. Adding a property ignore to the `Children` smallvec so it won't get serialized once it get's transformed into a Scene
2. Fixing WorldBuilder to not insert entities into the `Children` component
3. Changing children smallvec to `pub(create)` and adding a `impl Deref<Target = [Entity]> for Children`. If the used change this children component the side effects will be devastating

### PR DESCRIPTION
**Pull request by lassade**
Fixes #889,

TLDR;

The Problem
1. A entity is been added twice in `Children` component while spawning a scene
2. Performance takes a massive hit, `transform_propagate_system` took 13ms in release with a small skeleton tree ~20 bones
3. Affects gltf loading
4. Right now the only "correct" way of creating a `Scene` is using `WorldBuilder.with(Parent(...))` (still wrong for other reasons)

The Fix
1. Impl `FromResources` and `MapEntities` for `PrevousParent`
2. Checking for duplicates in the `Children` using a `debug_assert`
3. `Children.0` and `PreviousParent.0` are now `pub(crate)`, their contents are a reflect of `Parent`

Conclusion
This is the best short term fix, it will make so that the only true correct way of creating `Scene` will be `WorldBuilder.with_children` which has no caviats

### CODE REVIEW

**Comment by lassade** [comment]
@Moxinilian the CI error is very strange can you force a re-run to see if fixes?

**Comment by Moxinilian** [comment]
This is an issue with all CI currently, I don't have permission to do anything about it. I'll make sure cart is aware.

File: crates/bevy_transform/src/components/children.rs [hunk 1]
```diff
@@ -26,15 +26,9 @@ impl Children {
...
 impl Deref for Children {
-    type Target = SmallVec<[Entity; 8]>;
+    type Target = [Entity];
 
     fn deref(&self) -> &Self::Target {
-        &self.0
-    }
-}
-
-impl DerefMut for Children {
```
Comment (cart) [inline crates/bevy_transform/src/components/children.rs:36]: we need some way to re-order children. order already matters for things like UI, and once we have an editor it will matter for "scene organization".

I added this to enable those scenarios:

```rs
children.swap(a_index, b_index);
```

**Review by cart — COMMENTED** [review]
I think this is a reasonable short term fix. I made one small change (see comment below)
